tools/lsp/preview.rs

```diff
@@ -1066,27 +1068,53 @@ fn finish_parsing(preview_url: &Url, ok: bool) {
 }
 
 fn config_changed(config: PreviewConfig) {
-    if let Some(cache) = CONTENT_CACHE.get() {
-        let mut cache = cache.lock().unwrap();
-        if cache.config != config {
-            cache.config = config.clone();
+    let mut cache = CONTENT_CACHE.get_or_init(Default::default).lock().unwrap();
+    let in_flight = match cache.loading_state {
+        PreviewFutureState::Configuring(in_flight) => in_flight + 1,
```

Comment:
Instead of having an "in_flight", maybe there can be a single "NeedsReconfigure" state, and when set, we clean the cache before loading the preview.

Reply:
Users can change settings pretty quickly, quicker than the LS can parse all the files it knows. That might cause several configuration runs to be in flight. With no counter I can not know when the last is "done" at which point I should have a pretty sane state again.